FDB tests need to be modified in order to pass on setup with L2 configuration

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/Azure/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

on a setup with L2 configuration and Microsoft hwsku (BSL setup) FDB test are failing.
test_fdb.py failed because after configuring "sonic-clear fdb all" the fdb table should get clean but not all the entries get removed,
For a BSL DUT, all the ports are added into the same VLAN as L2 ports,  but on a T0 topology, the ports ‘connected’ to VMs are all L3 LAG interfaces, MAC learning is disabled.

So the packets sent from the Arista VMs will continually trigger mac learning on a BSL DUT, but not for a normal T0 DUT.

This is the reason why you see FDB test behaved differently on BSL and normal T0 testbed.    

So my conclusion is that, the “MAC address cannot be cleared issue”  is an issue of test case or test flow. The test case was designed for T0 topology, to run it on a different BSL configuration, some change has to be made.

I can suggest  two possible ways to make it compatible with BSL:

1. Change the VM configuration before the test and restore it afterward.
for the 4 relevant Arista VMs, I disabled the LLDP, BGP and port-channel on them, like below:
 
```

ARISTA03T1#config t

ARISTA03T1(config)#no router bgp

ARISTA03T1(config)#no lldp run

ARISTA03T1(config)#interface port-Channel 1

ARISTA03T1(config-if-Po1)#shutdown

ARISTA03T1(config-if-Po1)#exit

ARISTA03T1(config)#exit
```

And then run the FDB test, the test was passing.

2. Change the test case to have more accurate mac address verification,  only focus on the mac addresses generated by the test case itself, ignore the mac address triggered by other none relevant packets.   
 
**Steps to reproduce the issue:**

1. deploy t0 topo
2. deploy L2 on the switch with Microsoft HWSKU: https://github.com/Azure/SONiC/wiki/L2-Switch-mode
3. run the FDB tests fdb/test_fdb.py and receive the following results:
 ```
 def test_fdb(ansible_adhoc, ptfadapter, duthosts, rand_one_dut_hostname, ptfhost, pkt_type, toggle_all_simulator_ports_to_rand_selected_tor): 

 

        # Perform FDB clean up before each test and at the end of the final test 

>       fdb_cleanup(duthosts, rand_one_dut_hostname) 

 

ansible_adhoc = <function init_host_mgr at 0x7feae428eed0> 

duthosts   = <tests.common.devices.duthosts.DutHosts object at 0x7feae3fd0a90> 

pkt_type   = 'arp_reply' 

ptfadapter = <tests.common.plugins.ptfadapter.ptfadapter.PtfTestAdapter testMethod=runTest> 

ptfhost    = <tests.common.devices.ptf.PTFHost object at 0x7feae199df50> 

rand_one_dut_hostname = 'arc-switch1004' 

toggle_all_simulator_ports_to_rand_selected_tor = None 

 

fdb/test_fdb.py:192: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

 

duthosts = <tests.common.devices.duthosts.DutHosts object at 0x7feae3fd0a90> 

rand_one_dut_hostname = 'arc-switch1004' 

 

    def fdb_cleanup(duthosts, rand_one_dut_hostname): 

        """ cleanup FDB before and after test run """ 

        duthost = duthosts[rand_one_dut_hostname] 

        if fdb_table_has_no_dynamic_macs(duthost): 

            return 

        else: 

            duthost.command('sonic-clear fdb all') 

>           pytest_assert(wait_until(20, 2, fdb_table_has_no_dynamic_macs, duthost), "FDB Table Cleanup failed") 

E           Failed: FDB Table Cleanup failed 
```
**Describe the results you received:**

Actual output: "show mac" output on DUT after running "sonic-clear fdb all":

[u'  No.    Vlan  MacAddress         Port        Type',

 u'-----  ------  -----------------  ----------  -------',

 u'    1    1000  52:54:00:5D:F3:56  Ethernet32  Dynamic',

 u'    2    1000  52:54:00:73:1A:54  Ethernet8   Dynamic',

 u'    3    1000  52:54:00:AF:D4:66  Ethernet36  Dynamic',

 u'    4    1000  52:54:00:D2:4D:FD  Ethernet10  Dynamic',

 u'    5    1000  52:54:00:31:68:E3  Ethernet4   Dynamic',

 u'    6    1000  52:54:00:1E:A7:F8  Ethernet24  Dynamic',

 u'    7    1000  52:54:00:F9:8A:7F  Ethernet16  Dynamic',

 u'    8    1000  52:54:00:9D:26:DE  Ethernet6   Dynamic',

 u'    9    1000  52:54:00:E6:79:96  Ethernet20  Dynamic',

 u'   10    1000  52:54:00:0E:FB:60  Ethernet22  Dynamic',

 u'   11    1000  52:54:00:3F:6D:48  Ethernet28  Dynamic',

 u'Total number of entries 11']

 
**Describe the results you expected:**
Expected output: "show mac" output on DUT after running "sonic-clear fdb all" :

[u'No.    Vlan    MacAddress    Port    Type',

 u'-----  ------  ------------  ------  ------',

 u'Total number of entries 0']

**Additional information you deem important:**
<!--
software/ASIC/Hardware Flatform version and info
-->
    **Output of `show version`:**

`SONiC-OS-202012.131-01efb5454_Internal`


    **Attach debug file `sudo generate_dump`:**

    
[sonic_dump_arc-switch1025_20210721_124251.tar.gz](https://github.com/Azure/sonic-mgmt/files/6922763/sonic_dump_arc-switch1025_20210721_124251.tar.gz)

--- a/tests/fdb/test_fdb.py
+++ b/tests/fdb/test_fdb.py
@@ -163,7 +163,7 @@ def get_fdb_dynamic_mac_count(duthost):
     logger.info('"show mac" output on DUT:\n{}'.format(pprint.pformat(res['stdout_lines'])))
     total_mac_count = 0
     for l in res['stdout_lines']:
-        if "dynamic" in l.lower():
+        if "dynamic" in l.lower() and DUMMY_MAC_PREFIX in l.lower():
             total_mac_count += 1
     return total_mac_count
 

